app/master/cluster_master.py

```diff
@@ -219,7 +219,13 @@ def _handle_setup_failure_on_slave(self, slave):
         :type slave: Slave
         """
         internal_errors.labels(ErrorType.SetupBuildFailure).inc()  # pylint: disable=no-member
-        raise BadRequestError('Setup failure handling on the master is not yet implemented.')
+        build = self.get_build(slave.current_build_id)
+        build.setup_failures += 1
```

Comment:
I'm pretty sure this method is only ever called from tornado's thread, so I don't have to worry about locking on setup failures before incrementing.

I saw the Counter class in `app/util`. I could use that instead.